hnn_core/gui/gui.py

```diff
@@ -26,15 +26,307 @@
+                                                   ('MPI', 'MPI')],
+                                          value='Joblib',
+                                          description='Backend:')
+
+        # visualization layout
+        self.viz_layout_selection = Dropdown(options=[('Horizontal', 'L-R'),
+                                                      ('Vertical', 'U-D')],
+                                             value='L-R',
+                                             description='Layout:')
+
+        self.mpi_cmd = Text(value='mpiexec',
+                            placeholder='Fill if applies',
+                            description='MPI cmd:',
+                            disabled=False)
+
+        self.n_jobs = BoundedIntText(value=1, min=1,
+                                     max=multiprocessing.cpu_count(),
+                                     description='Cores:', disabled=False)
+
+        self.drive_type_selection = RadioButtons(
+            options=['Evoked', 'Poisson', 'Rhythmic'],
+            value='Evoked',
+            description='Drive:',
+            disabled=False,
+            layout=Layout(width=self.drive_widget_width))
+
+        self.location_selection = RadioButtons(
+            options=['proximal', 'distal'], value='proximal',
+            description='Location', disabled=False,
+            layout=Layout(width=self.drive_widget_width))
+
+        self.add_drive_button = create_expanded_button(
+            'Add drive', 'primary', height='30px',
+            button_color=self.theme_color)
+
+        # Run, delete drives and load button
+        self.run_button = create_expanded_button('Run', 'success',
+                                                 height='30px',
+                                                 button_color=self.theme_color)
+
+        self.load_button = FileUpload(accept='.json,.param',
+                                      multiple=False,
+                                      style={'button_color': self.theme_color},
+                                      description='Load network',
+                                      button_style='success')
+
+        self.delete_drive_button = create_expanded_button(
+            'Delete drives', 'success', height='30px',
+            button_color=self.theme_color)
+
+        # Visualization figure list
+        self.plot_outputs_list = list()
+        self.plot_dropdowns_list = list()
+        # Add drive section
+        self.drive_widgets = list()
+        self.drive_boxes = list()
+
+        self._init_ui_components()
+
+    def _init_ui_components(self):
+        """Initialize larger UI components and dynamical output windows. It's
+        not encouraged for users to modify or access attributes in this part.
+        """
+        self.connectivity_sliders = self.init_cell_connectivity(self.params)
+        # dynamic larger components
+        self.drives_out = Output()  # window to add new drives
+        self.log_out = Output(
```

Comment:
I tend to not to do so because almost all attributes in this method are  not supposed to be accessible to users. I'll mark then as private with `_`prefix

Reply:
Just declare everything in init, whether private or not. Also init should have minimal logic

See: https://betterprogramming.pub/6-best-practices-for-defining-the-initialization-method-in-python-9173808bfb8f.